fixtures/cognitive-complexity.go

```diff
@@ -0,0 +1,277 @@
+// Test of cognitive complexity.
+
+// Package pkg ...
+package pkg
+
+import (
+	"fmt"
+	ast "go/ast"
+	"log"
+	"testing"
+
+	"github.com/blang/semver"
+	"k8s.io/klog"
+)
+
+// Test IF and Boolean expr
+func f(x int) bool { // MATCH /function f has cognitive complexity 3 (> max enabled 0)/
+	if x > 0 && true || false { // +3
+		return true
+	} else {
+		log.Printf("non-positive x: %d", x)
+	}
+	return false
+}
+
+// Test IF
+func g(f func() bool) string { // MATCH /function g has cognitive complexity 1 (> max enabled 0)/
+	if ok := f(); ok { // +1
+		return "it's okay"
+	} else {
+		return "it's NOT okay!"
+	}
+}
+
+// Test Boolean expr
+func h(a, b, c, d, e, f bool) bool { // MATCH /function h has cognitive complexity 2 (> max enabled 0)/
+	return a && b && c || d || e && f //FIXME: complexity should be 3?
```

Comment:
Yes, I think it should be 3 as well:

>Cognitive complexity increments for each new sequence of like operators

Reply:
Doubt: 
`a && !(b || c)` 
`a && !(b && c)`

have both he same complexity (2)?

Reply:
I re-wrote the computation of complexity of binary expressions. Now it is in line with the definition

Reply:
>Doubt: 
>`a && !(b || c)` 
>`a && !(b && c)`
>
>have both he same complexity (2)?

I think so.